softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractFactory.java

```diff
@@ -0,0 +1,35 @@
+package com.softgroup.common.router.api;
+
+import com.softgroup.common.protocol.Request;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 27.02.17.
+ */
+
+public abstract class AbstractFactory<T extends Handler> {
```

Comment:
@AlexKazmirchuk Where is interface for factory? Factory must be injected by interface.